Author: Mauricio Faria de Oliveira <mauricfo@linux.vnet.ibm.com>
Description: prod multipathd to actually use -B/bindings_read_only on discovery
Last-Update: 2016-01-07

---
 multipathd/main.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/multipathd/main.c
+++ b/multipathd/main.c
@@ -1745,6 +1745,7 @@
 #endif
 	int rc, pid_rc;
 	char *envp;
+	int bindings_read_only;
 
 	mlockall(MCL_CURRENT | MCL_FUTURE);
 	sem_init(&exit_sem, 0, 0);
@@ -1770,9 +1771,19 @@
 	condlog(2, "--------start up--------");
 	condlog(2, "read " DEFAULT_CONFIGFILE);
 
+	/*
+	 * The conf->bindings_read_only value is reset
+	 * in load_config().  Save it, since discovery
+	 * (which exercises the path for this setting)
+	 * happens after this point.
+	 */
+	bindings_read_only = conf->bindings_read_only;
+
 	if (load_config(DEFAULT_CONFIGFILE, udev))
 		goto failed;
 
+	conf->bindings_read_only = bindings_read_only;
+
 	dm_drv_version(conf->version, TGT_MPATH);
 	if (init_checkers()) {
 		condlog(0, "failed to initialize checkers");
